pkg/cmd/roachtest/cluster.go

```diff
@@ -955,7 +957,14 @@ func (f *clusterFactory) newCluster(
 
 		l.PrintfCtx(ctx, "Attempting cluster creation (attempt #%d/%d)", i, maxAttempts)
 		createVMOpts.ClusterName = c.name
-		err = create(ctx, l, cfg.username, cfg.spec.NodeCount, createVMOpts, providerOptsContainer)
+		opts := []cloud.ClusterCreateOpts{{Nodes: cfg.spec.NodeCount, CreateOpts: createVMOpts, ProviderOptsContainer: providerOptsContainer}}
```

Comment:
If we take the approach of not using the number of nodes from the cluster count, the if condition can be an append. with `cfg.spec.WorkloadNode` as the Nodes value.